### PR TITLE
Use head definition meta to warn about unused args

### LINKED ISSUE #8184: Wrong line no. in "warning: default arguments in ... are never used"
```
warning: default arguments in re/2 are never used
  lib/ex_doc/formatter/html/autolink.ex:859
```

The mentioned line is the line of the last clause for that function
https://github.com/eksperimental/ex_doc/blob/7b4d1ee25d2bed02a2aa9177d71257fd105d0a85/lib/ex_doc/formatter/html/autolink.ex#L859

but the actual offending line is the function head.
https://github.com/eksperimental/ex_doc/blob/7b4d1ee25d2bed02a2aa9177d71257fd105d0a85/lib/ex_doc/formatter/html/autolink.ex#L747

maybe using a function head is the culprit

### PR DESCRIPTION
**Pull request by ggcampinho**
Fixes #8184

### CODE REVIEW

File: lib/elixir/lib/module/locals_tracker.ex [hunk 1]
```diff
@@ -122,24 +122,28 @@ defmodule Module.LocalsTracker do
...
+    collect_warnings({tuple, kind, {meta, meta}, default}, acc, reachable)
+  end
+
+  defp collect_warnings({tuple, kind, {meta, head_meta}, default}, acc, reachable) do
     {name, arity} = tuple
     min = arity - default
     max = arity
 
     case min_reachable_default(max, min, :none, name, reachable) do
       :none -> [{meta, {:unused_def, tuple, kind}} | acc]
```
Comment (ggcampinho) [inline lib/elixir/lib/module/locals_tracker.ex]: I guess I could change all to use the head meta, but I'm not sure what is best

Comment (josevalim) [reply lib/elixir/lib/module/locals_tracker.ex]: Let's go with head meta  everywhere and see how it impacts the existing suite and see if the warnings get worse or better.

File: lib/elixir/test/elixir/kernel/warning_test.exs [hunk 2]
```diff
@@ -352,7 +352,7 @@ defmodule Kernel.WarningTest do
                defp hello(1), do: :ok
              end
              """)
-           end) =~ "function hello/1 is unused"
+           end) =~ "function hello/1 is unused\n  nofile:3"
```
[the remark this answers is not in the document]
Comment (josevalim) [reply lib/elixir/test/elixir/kernel/warning_test.exs]: Should we fix this one too then?

[the remark this answers is not in the document]
Comment (ggcampinho) [reply lib/elixir/test/elixir/kernel/warning_test.exs]: Done

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**